backend/tracim_backend/applications/upload_permissions/lib.py

```diff
@@ -0,0 +1,284 @@
+        self.save(upload_permission=upload_permission_to_disable)
+        return upload_permission_to_disable
+
+    def frontend_url(self, upload_permission: UploadPermission) -> str:
+        frontend_ui_base_url = get_frontend_ui_base_url(config=self._config)
+        return FRONTEND_UPLOAD_PERMISSION_LINK_PATTERN.format(
+            frontend_ui_base_url=frontend_ui_base_url,
+            upload_permission_token=upload_permission.token,
+        )
+
+    def save(self, upload_permission: UploadPermission) -> UploadPermission:
+        self._session.add(upload_permission)
+        return upload_permission
+
+    def _notify_uploaded_contents(
+        self,
+        uploader_username: str,
+        workspace: Workspace,
+        uploaded_contents: typing.List[ContentInContext],
+    ):
+        email_manager = self._get_email_manager(self._config, self._session)
+        # TODO - G.M - 2019-08-12 - handle exceptions there
+        email_manager.notify_new_upload(
+            uploaded_contents=uploaded_contents,
+            uploader_username=uploader_username,
+            workspace=workspace,
+        )
+
+    def upload_files(
+        self,
+        upload_permission: UploadPermission,
+        uploader_username: str,
+        message: str,
+        files: typing.List[cgi.FieldStorage],
+        do_notify: bool = False,
+    ):
+        content_api = ContentApi(
+            config=self._config, current_user=upload_permission.author, session=self._session
+        )
+        _ = content_api.translator.get_translation
+        folder_label = _("Files uploaded by {username} on {date}").format(
+            username=uploader_username, date=datetime.now()
+        )
+        upload_folder = content_api.create(
+            content_type_slug=content_type_list.Folder.slug,
+            workspace=upload_permission.workspace,
+            label=folder_label,
+            do_notify=False,
+            do_save=True,
+            content_namespace=ContentNamespaces.UPLOAD,
+        )
+        created_contents = []
+        for _file in files:
+            content = content_api.create(
+                filename=_file.filename,
+                content_type_slug=content_type_list.File.slug,
+                workspace=upload_permission.workspace,
+                parent=upload_folder,
+                do_notify=False,
+                content_namespace=ContentNamespaces.UPLOAD,
+            )
+            content_api.save(content, ActionDescription.CREATION)
+            with new_revision(session=self._session, tm=transaction.manager, content=content):
+                content_api.update_file_data(
+                    content,
+                    new_filename=_file.filename,
+                    new_mimetype=_file.type,
+                    new_content=_file.file,
+                )
+            content_api.create_comment(
+                parent=content,
+                content=_("Message from {username}: {message}").format(
```

Comment:
`message` can be empty string ? If yes, maybe no comment or different comment ?

Reply:
empty comment now supported with specific message.